community/shell/src/main/java/org/neo4j/shell/StartClient.java

```diff
@@ -156,8 +181,19 @@ void start( String[] arguments, CtrlCHandler signalHandler )
         String port = args.get( ARG_PORT, null );
         String name = args.get( ARG_NAME, null );
         String pid = args.get( ARG_PID, null );
+        boolean ver = args.getBoolean( ARG_VERSION, false, true );
 
-        if ( (path != null && (port != null || name != null || host != null || pid != null))
+        if ( ver )
+        {
+            String version = Version.getKernelVersion();
+            String edition = "Community";
+            if ( factory.getClass() != GraphDatabaseFactory.class )
```

Comment:
`GraphDatabaseFactory` can also have `String #getEdition()` method to avoid class name check